bcda/api.go

```diff
@@ -123,12 +123,24 @@ func jobStatus(w http.ResponseWriter, r *http.Request) {
 
 		jsonData, err := json.Marshal(rb)
 		if err != nil {
+			errorStatus := createOperationOutcome("marshalling error")
+			_, err = w.Write([]byte(errorStatus))
+			if err != nil {
```

Comment:
Are lines 128-130 redundant?  Looks like we are doing the following regardless:
`http.Error(w,` http.StatusText(500), 500)
return`

Reply:
This is also true, it is a bit redundant. So basically we marshal the data but if an error happens we have to write the operation outcome response. But the writing of that response error could also fail so I try to capture that as well. Kinda stuck in this error handling predicament but would love a new design here.